staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go

```diff
@@ -288,8 +289,8 @@ func TestGet(t *testing.T) {
 			if err != nil {
 				t.Fatalf("Get failed: %v", err)
 			}
-			if !reflect.DeepEqual(tt.expectedOut, out) {
-				t.Errorf("pod want=\n%#v\nget=\n%#v", tt.expectedOut, out)
+			if diff := cmp.Diff(tt.expectedOut, out); diff != "" {
```

Comment:
Hrm... I kind of expected this to be using reflect.DeepEqual and then printing the diff, not delegating to `cmp.Diff` to actually do the diff (since the semantic before was the objects are reflect.DeepEqual, not cmp.Diff equal).  Is there an argument elsewhere that we do this?

Reply:
I.e. https://github.com/stevekuznetsov/kubernetes/blob/2c973c7db3805e3d58b3dbdaf710d60c4e4f80e9/vendor/github.com/google/go-cmp/cmp/compare.go#L92 is *close* to `reflect.DeepEqual`, but not the same.  If someone adds `Equal` to a struct, cmp.Diff might return true and DeepEqual would not.

Reply:
I'd say two things:

1. these are fake objects without semantic meaning in the API, so they do not need any special handling (e.g. semantic equality) - this patch does not add a custom handler, and therefore does not change the actual comparison being done
2. using one check for comparison and one for the diff printing in my experience is annoying to the user when the two differ

Reply:
I agree with clayton that we should not relax the test-failing check from reflect.DeepEqual to cmp.Diff.

Reply:
OK, happy to use `cmp.Diff` only for printing.